### ISSUE TITLE
Please consider adding an example for checking values against types

### ISSUE THREAD

**Issue by Etheryte** [typescript-rtti/typescript-rtti#50]
The main use case I'm monitoring this project and other similar ones is to eventually replace [typanion](https://github.com/arcanis/typanion) and other similar options with something that doesn't require separate annotations.  

So far typescript-rtti looks very promising, it seems it already correctly work for interfaces:  

```ts
import { reflect } from 'typescript-rtti';

interface Foo {
    foo: number;
};

const a: unknown = {
    foo: 123,
};
const b: unknown = {
    bar: 123,
};

reflect<Foo>().matchesValue(a); // true;
reflect<Foo>().matchesValue(b); // false
```

A few questions and/or suggestions:  

 - Is this the right way to check an arbitrary value against an interface? It would be handy to add a canonical example of the right way to the readme.
 - Is it also possible to do this for `type Foo = { ... }`? Right now I couldn't get it to work, but I suspect it might simply be that I don't know the library well enough.

**Comment by rezonant** [comment]
> Is this the right way to check an arbitrary value against an interface? It would be handy to add a canonical example of the right way to the readme.

Yes, this is correct for checking an arbitrary object against a type reference for compatibility.

> Is it also possible to do this for type Foo = { ... }? Right now I couldn't get it to work, but I suspect it might simply be that I don't know the library well enough.

It would be the right way yes. There is one missing type reference type to enable it: Object literals

It's tricky to represent this within the current setup, but not impossible. It would require either duplicating a lot of logic from ReflectedClass, or abstracting it out so it can be shared. This is definitely a 1.0 enhancement.

**Comment by Etheryte** [comment]
Thank you very much for your input, and for your work on this. This looks very promising, we're definitely going to try adopting it when it works with types as well.

**Comment by rezonant** [comment]
OK provided all the test passes look good, this will be included in the next release (0.5.5)

**Comment by rezonant** [comment]
OK, 0.5.5 is released with this functionality. Try it out here:
https://typescript-rtti.org/#aW1wb3J0IHsgcmVmbGVjdCB9IGZyb20gJ3R5cGVzY3JpcHQtcnR0aSc7Cgp0eXBlIEYgPSB7IGZvbzogMTIzLCBiYXI6IHN0cmluZyB9Cgpjb25zb2xlLmxvZyhyZWZsZWN0PEY+KCkubWF0Y2hlc1ZhbHVlKHsgZm9vOiAxMjMsIGJhcjogMTIzIH0pKTs=
